### ISSUE TITLE
Xournal uses up all the power of one cpu core, thus resulting in lag and crush 

### ISSUE THREAD

**Issue by enrico223** [xournalpp/xournalpp#3405]
**Affects versions :**
 - OS: Fedora 34 
 - Gnome-Wayland
 - libgtk 3.24.30
 - Version of Xournal++: 1.1.0
 - Installation method: official release from fedora repositories

**Describe the bug**
Xournal uses up all the power of one cpu core, thus resulting in lag and crash 

**To Reproduce**
1. Open a pdf file 
2. Annotate it with pen and highlighter 
3. try to zoom and move the focus around 

**Expected behavior**
Not crashing 

**Additional context**
I have a HP Spectre x360 that I use with the stylus, all the annotations are made with it.

**Comment by Technius** [comment]
Can you provide terminal output or a crash log from `~/.cache/xournalpp/errorlogs`?

**Comment by enrico223** [comment]
Unfortunately there is no creah log from that directory. Idk why. When Xournal sort of crashes, I get a message with the alternatives to either close the app or wait. In both cases, no log. What do you mean by terminal output? A screenshot of htop?

**Comment by Technius** [comment]
No, I mean running xournalpp from command line.

I think you may be encountering the "zoom fit hang with different sized pages" issue (#1660, #2770). Do you still get the hang/crash with zoom fit disabled?

**Comment by enrico223** [comment]
Yes, i still get it. Here the screen when I launch it from the command line and make it hang. 
![Screenshot from 2021-09-21 00-08-39](https://user-images.githubusercontent.com/72576597/134082650-a6a067c0-b23b-4795-870e-e6a0ad313c6b.png)

**Comment by Technius** [comment]
Seems like a new bug then. Are you able to upload the PDF you have so it's easier for us to debug?

**Comment by enrico223** [comment]
[J. R. Gillingham_ppdf. 55_82_W1_required.pdf](https://github.com/xournalpp/xournalpp/files/7199090/J.R.Gillingham_ppdf.55_82_W1_required.pdf)
It's all yours, I hope it help!

**Comment by enrico223** [comment]
Is that pdf useful? do you need any other info?

**Comment by LittleHuba** [comment]
Just did a quick test. This will be a hard one to fix. The issue is that the current rendering system renders the full page independent on if it is shown or not. From this full page the actual viewport is the shown within the graphical user interface. Because your PDF has very big images for each page, rendering takes its time and zooming leads to a backlog of unrendered resolutions. If you wait long enough, the backlog is eventually worked off and the program resumes normal working behavior.

Proposed steps to fix this:

1. We can take a look at the rendering system to see if we can reduce the backlog more efficiently. For example, by removing rendering jobs for other resolutions than the current one, whenever we add a new rendering job.
2. Fix the rendering system to only render the part of the page that is currently shown (don't know how far this is already handled)

**Comment by enrico223** [comment]
Oh, all right. Thank you, I appreciate the feedback!

**Comment by Technius** [comment]
There was #2982 that @personalizedrefrigerator was working on a while back as a potential optimization, which I think would work well for the problem you've described here.

**Comment by enrico223** [comment]
@Technius I came back to use xournal with pdfs of a size comparable to the one I had problem with, and now I do not experience any crashes (i'm on fedora 37 now). 
I don't know what you guys did, but thank you!

**Comment by Technius** [comment]
Are you using the nightly build? Several optimization changes were made in the nightly version recently.

**Comment by enrico223** [comment]
I don't think so, the version is the one shipped in Fedora 37 `xournal-0.4.8.2016-11.fc37.x86_64`

**Comment by rolandlo** [comment]
@enrico223 So you are now using legacy Xournal, not Xournal++. That's a different project.

**Comment by enrico223** [comment]
@rolandlo Also using `xournalpp 1.1.3-1.fc37 `, I am not experiencing any problem
